pruntime/enclave/src/contracts/balance.rs

```diff
@@ -123,4 +198,31 @@ impl contracts::Contract<Command, Request, Response> for Balance {
             Ok(resp) => resp
         }
     }
+
+    fn handle_event(&mut self, ce: chain::Event) {
+        if let chain::Event::pallet_phala(pe) = ce {
+            if let phala::RawEvent::TransferToTee(who, amount) = pe {
+                println!("TransferToTee from :{:?}, {:}", who, amount);
+                let account_id = chain::AccountId::decode(&mut who.as_slice()).unwrap();
+                let dest = AccountIdWrapper(account_id);
+                println!("dest:{:?}", dest.clone());
+                if let Some(dest_amount) = self.accounts.get_mut(&dest) {
+                    let dest0 = *dest_amount;
+                    *dest_amount += amount;
+                    println!("{:?}'s balance:{}", dest, *dest_amount);
+                } else {
+                    self.accounts.insert(dest, amount);
+                }
+            } else if let phala::RawEvent::TransferToChain(id, who, amount, sequence) = pe {
+                println!("TransferToChain id : {:?}, who: {:?}, amount: {:}", id, who, amount);
+                let mut id32 = [0u8; 32];
+                id32.copy_from_slice(&id);
+                let account_id = chain::AccountId::decode(&mut who.as_slice()).unwrap();
+                let transfer_data = TransferData { id: id32, dest: AccountIdWrapper(account_id), amount, signature: None, sequence };
+                println!("transfer data:{:?}", transfer_data);
+                self.queue.retain(|x| x.id != transfer_data.id);
```

Comment:
Because the events come by an order of increasing `sequence`, we can simply remove all the messages in the queue where their sequence id is less than or equal to the current one